erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -905,14 +904,35 @@ def get_items(self):
 
 	def set_scrap_items(self):
 		if self.purpose != "Send to Subcontractor" and self.purpose in ["Manufacture", "Repack"]:
-			scrap_item_dict = self.get_bom_scrap_material(self.fg_completed_qty)
+			scrap_item_dict = {}
+			if frappe.db.get_single_value("Manufacturing Settings",	"backflush_scrap_items_based_on") =="BOM":
+				scrap_item_dict = self.get_bom_scrap_material(self.fg_completed_qty)
+			else:
```

Comment:
Since this is a Select field, I'd say make this check more explicit for Job Card